### PR TITLE
Refactoring: Remove the "erased" attribute of Instance

### PR DESCRIPTION
**Pull request by JukkaL**
It's not used for anything.

### CODE REVIEW

File: mypy/types.py [hunk 1]
```diff
@@ -1032,16 +1032,13 @@ def try_getting_instance_fallback(typ: ProperType) -> Optional[Instance]:
     __slots__ = ('type', 'args', 'erased', 'invalid', 'type_ref', 'last_known_value')
```
Comment (97littleleaf11) [inline mypy/types.py:1032]: Remove it from slots？

Comment (JukkaL) [reply mypy/types.py]: Done!

File: mypy/checkmember.py [hunk 2]
```diff
@@ -847,7 +847,7 @@ def analyze_enum_class_attribute_access(itype: Instance,
     # in typeshed. However, this is really more of an implementation detail of how Enums
     # are typed, and we really don't want to treat every single Enum value as if it were
     # from type variable substitution. So we reset the 'erased' field here.
-    return itype.copy_modified(erased=False, last_known_value=enum_literal)
```
Comment (97littleleaf11) [inline mypy/checkmember.py:850]: Docstring should be modified too

Comment (JukkaL) [reply mypy/checkmember.py:850]: Removed the comment.

File: mypy/expandtype.py [hunk 3]
```diff
@@ -94,8 +94,7 @@ def visit_type_var(self, t: TypeVarType) -> Type:
         if isinstance(repl, Instance):
             inst = repl
             # Return copy of instance with type erasure flag on.
```
Comment (AlexWaygood) [inline mypy/expandtype.py:96]:
```suggestion
            # Return copy of instance.
```

File: mypy/expandtype.py [hunk 4]
```diff
@@ -104,8 +103,7 @@ def visit_param_spec(self, t: ParamSpecType) -> Type:
...
```
Comment (AlexWaygood) [inline mypy/expandtype.py:105]:
```suggestion
            # Return copy of instance.
```

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**